runner-macros/src/entry.rs

```diff
@@ -0,0 +1,148 @@
+use proc_macro::TokenStream;
+use proc_macro2::Span;
+use quote::{quote, quote_spanned, ToTokens};
+
+enum Flavor {
+    Sandbox,
+    Testnet,
+    Mainnet,
+}
+
+fn parse_knobs(
+    mut input: syn::ItemFn,
+    args: syn::AttributeArgs,
+    is_test: bool,
+) -> Result<TokenStream, syn::Error> {
+    if input.sig.asyncness.take().is_none() {
+        let msg = "the `async` keyword is missing from the function declaration";
+        return Err(syn::Error::new_spanned(input.sig.fn_token, msg));
+    }
+
+    let mut flavor = None;
+    for arg in args {
+        match arg {
+            syn::NestedMeta::Meta(syn::Meta::NameValue(namevalue)) => {
+                let ident = namevalue
+                    .path
+                    .get_ident()
+                    .ok_or_else(|| {
+                        syn::Error::new_spanned(&namevalue, "Must have specified ident")
+                    })?
+                    .to_string()
+                    .to_lowercase();
```

Comment:
why convert to lowercase if this branch will always error?

Reply:
oops, thanks for noticing this. This macro was mainly copypasta'd over from tokio